blockdag/utxoio.go

```diff
@@ -19,26 +19,25 @@ import (
 // 	hasDiffChild | Boolean   | Indicates if a diff child exist
 //  diffChild    | Hash      | The diffChild's hash. Empty if hasDiffChild is true.
 //  diff		 | UTXODiff  | The diff data's diff
-func serializeBlockUTXODiffData(diffData *blockUTXODiffData) ([]byte, error) {
-	w := &bytes.Buffer{}
+func serializeBlockUTXODiffData(buffer *bytes.Buffer, diffData *blockUTXODiffData) error {
```

Comment:
I think we want any function that doesn't need the buffer as bytes.Buffer to get it as io.Writer